src/overflow_check_conditional.rs

```diff
@@ -0,0 +1,41 @@
+use rustc::lint::*;
+use syntax::ast::*;
+use utils::{span_lint, span_help_and_lint, snippet, span_lint_and_then};
+
+/// **What it does:** This lint finds classic overflow checks.
+///
+/// **Why is this bad?** Most classic overflow checks would cause the compiler to panic.
```

Comment:
I would reformulate that to insist on the fact that while those are classic in C, they won’t work in Rust. Also point out that Rust does not need them because of the `overflowing_*` and `wrapping_*` family of functions.